### PR TITLE
[exporter/datadog] Include retry count in retrier log

### PR DESCRIPTION
**Pull request by songy23**
**Description:** <Describe what has changed.>
Include retry count in retrier log in datadogexporter

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Comment by songy23** [comment]
Please add `Skip Changelog`

File: exporter/datadogexporter/internal/clientutil/retrier.go [hunk 1]
```diff
@@ -81,7 +81,7 @@ func (r *Retrier) DoWithRetries(ctx context.Context, fn func(context.Context) er
 
 		backoffDelayStr := backoffDelay.String()
 		r.logger.Info(
-			"Request failed. Will retry the request after interval.",
+			fmt.Sprintf("Request failed with retriable errors. Will retry the request after interval. Retry attempts: %d", retryNum),
```
Comment (dineshg13) [inline exporter/datadogexporter/internal/clientutil/retrier.go:84]: Can we use `zap.Int`

Comment (songy23) [reply exporter/datadogexporter/internal/clientutil/retrier.go]: Sure, updated

**Review by dineshg13 — APPROVED** [review]

**Comment by songy23** [comment]
This PR is ready to merge

**Review by mx-psi — APPROVED** [review]

**Pull request merged**